src/rgw/rgw_rest.cc

```diff
@@ -655,7 +655,9 @@ void end_header(struct req_state* s, RGWOp* op, const char *content_type,
       s->formatter->close_section();
     }
     s->formatter->output_footer();
-    dump_content_length(s, s->formatter->get_len());
+    if (s->obj_size) {
```

Comment:
The content length here is that of the end header's and its not relevant in this context. For range requests on zero sized objects, without suppressing this header, the output looks something like this:

HTTP/1.1 416 Requested Range Not Satisfiable
Content-Range: bytes */0
Content-Length: 0
Accept-Ranges: bytes
X-Trans-Id: tx000000000000000000026-00579f3001-1011-default
Content-Length: 12
Accept-Ranges: bytes
Content-Type: text/plain; charset=utf-8
Date: Mon, 01 Aug 2016 11:18:25 GMT